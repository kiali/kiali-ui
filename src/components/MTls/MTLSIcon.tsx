import * as React from 'react';
import { Tooltip, TooltipPosition } from '@patternfly/react-core';
import fullIcon from '../../assets/img/mtls-status-full.svg';
import hollowIcon from '../../assets/img/mtls-status-partial.svg';
import fullIconDark from '../../assets/img/mtls-status-full-dark.svg';
import hollowIconDark from '../../assets/img/mtls-status-partial-dark.svg';

export { fullIcon, hollowIcon, fullIconDark, hollowIconDark };

type Props = {
  icon: string;
  iconClassName: string;
  tooltipText: string;
  tooltipPosition: TooltipPosition;
};

export enum MTLSIconTypes {
  LOCK_FULL = 'LOCK_FULL',
  LOCK_HOLLOW = 'LOCK_HOLLOW',
  LOCK_FULL_DARK = 'LOCK_FULL_DARK',
  LOCK_HOLLOW_DARK = 'LOCK_HOLLOW_DARK'
}

const nameToSource = new Map<string, string>([
  [MTLSIconTypes.LOCK_FULL, fullIcon],
  [MTLSIconTypes.LOCK_FULL_DARK, fullIconDark],
  [MTLSIconTypes.LOCK_HOLLOW, hollowIcon],
  [MTLSIconTypes.LOCK_HOLLOW_DARK, hollowIconDark]
]);

class MTLSIcon extends React.Component<Props> {
  render() {
    return (
      <Tooltip
        aria-label={'mTLS status'}
        position={this.props.tooltipPosition}
        enableFlip={true}
        content={this.props.tooltipText}
      >
        <img
          className={this.props.iconClassName}
          src={nameToSource.get(this.props.icon)}
          alt={this.props.tooltipPosition}
        />
      </Tooltip>
    );
  }
}
export default MTLSIcon;
