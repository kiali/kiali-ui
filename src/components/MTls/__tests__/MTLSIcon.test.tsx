import * as React from 'react';
import { shallow } from 'enzyme';
import MTLSIcon, { MTLSIconTypes } from '../MTLSIcon';
import { shallowToJson } from 'enzyme-to-json';

const mockIcon = (icon: string) => {
  const component = (
    <MTLSIcon icon={icon} iconClassName={'className'} overlayText={'Overlay Test'} overlayPosition={'left'} />
  );
  return shallow(component);
};

describe('when Icon is LOCK_FULL', () => {
  it('MTLSIcon renders properly', () => {
    const wrapper = mockIcon(MTLSIconTypes.LOCK_FULL);

    expect(shallowToJson(wrapper)).toBeDefined();
    expect(shallowToJson(wrapper)).toMatchSnapshot();

    expect(wrapper.name()).toEqual('OverlayTrigger');
    expect(wrapper.props().placement).toEqual('left');

    expect(wrapper.children()).toBeDefined();
    expect(wrapper.children().name()).toEqual('img');
    expect(wrapper.children().prop('className')).toEqual('className');
    expect(wrapper.children().prop('src')).toEqual('mtls-status-full.svg');

    const overlay = wrapper.props().overlay;
    expect(overlay.props.bsClass).toEqual('tooltip');
    expect(overlay.props.placement).toEqual('right');
    expect(overlay.props.children).toContain('Overlay Test');
  });
});

describe('when Icon is LOCK_HOLLOW', () => {
  it('MTLSIcon renders properly', () => {
    const wrapper = mockIcon(MTLSIconTypes.LOCK_HOLLOW);
    expect(wrapper.children().prop('src')).toEqual('mtls-status-partial.svg');
  });
});
