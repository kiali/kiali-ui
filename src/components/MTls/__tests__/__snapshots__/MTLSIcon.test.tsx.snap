// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when Icon is LOCK_FULL MTLSIcon renders properly 1`] = `
<OverlayTrigger
  defaultOverlayShown={false}
  overlay={
    <Tooltip
      bsClass="tooltip"
      id="mtls-status-masthead"
      placement="right"
    >
      Overlay Test
    </Tooltip>
  }
  placement="left"
  rootClose={false}
  trigger={
    Array [
      "hover",
      "focus",
    ]
  }
>
  <img
    alt="left"
    className="className"
    src="mtls-status-full.svg"
  />
</OverlayTrigger>
`;
